File: app/src/main/java/sfllhkhan95/doodle/core/views/ColorPicker.java
package sfllhkhan95.doodle.core.views;

import android.content.Context;

import sfllhkhan95.doodle.core.utils.OnColorPickedListener;
import yuku.ambilwarna.AmbilWarnaDialog;

public class ColorPicker implements AmbilWarnaDialog.OnAmbilWarnaListener {

    private final AmbilWarnaDialog dialog;

    private OnColorPickedListener onColorPickedListener;

    public ColorPicker(Context context, int color) {
        this.dialog = new AmbilWarnaDialog(context, color, true, this);
    }

    @Override
    public void onCancel(AmbilWarnaDialog dialog) {

    }

    @Override
    public void onOk(AmbilWarnaDialog dialog, int color) {
        if (onColorPickedListener != null) {
            onColorPickedListener.onColorPicked(color);
        }
    }

    public void setOnColorPickedListener(OnColorPickedListener onColorPickedListener) {
        this.onColorPickedListener = onColorPickedListener;
    }

    public void show() {
        this.dialog.show();
    }
}

File: app/src/main/java/sfllhkhan95/doodle/core/views/ColorPicker.kt
package sfllhkhan95.doodle.core.views

import android.content.Context
import android.support.v7.app.AlertDialog
import com.rarepebble.colorpicker.ColorObserver
import com.rarepebble.colorpicker.ColorPickerView
import com.rarepebble.colorpicker.ObservableColor
import sfllhkhan95.doodle.core.utils.OnColorPickedListener

open class ColorPicker(val context: Context, color: Int, val dialogTheme: Int) : ColorObserver {

    protected val picker: ColorPickerView = ColorPickerView(context)

    init {
        picker.color = color
        picker.showAlpha(true)
        picker.showHex(false)
        picker.showPreview(true)
        picker.addColorObserver(this)
    }

    private var onColorPickedListener: OnColorPickedListener? = null

    override fun updateColor(observableColor: ObservableColor?) {
        observableColor?.let {
            if (onColorPickedListener != null) {
                onColorPickedListener!!.onColorPicked(it.color)
            }
        }
    }

    fun setOnColorPickedListener(onColorPickedListener: OnColorPickedListener) {
        this.onColorPickedListener = onColorPickedListener
    }

    fun show() {
        AlertDialog.Builder(context, dialogTheme).setView(picker).show()
    }
}